ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/RateLimitingInterceptorSpec.scala

```diff
@@ -217,6 +223,126 @@ final class RateLimitingInterceptorSpec
     }
   }
 
+  it should "limit the number of streams" in {
+
+    val metrics = new Metrics(new MetricRegistry)
+
+    val limitStreamConfig = RateLimitingConfig.Default.copy(maxStreams = 2)
+
+    val waitService = new WaitService()
+    withChannel(metrics, waitService, limitStreamConfig).use { channel: Channel =>
+      {
+        for {
+          fStatus1 <- streamHello(channel) // Ok
+          fStatus2 <- streamHello(channel) // Ok
+          activeStreams = metrics.daml.lapi.streams.active.getCount
+          fStatus3 <- streamHello(channel) // Limited
+          status3 <- fStatus3 // Closed as part of limiting
+          _ = waitService.completeStream()
+          status1 <- fStatus1
+          fStatus4 <- streamHello(channel) // Ok
+          _ = waitService.completeStream()
+          status2 <- fStatus2
+          _ = waitService.completeStream()
+          status4 <- fStatus4
+        } yield {
+          activeStreams shouldBe limitStreamConfig.maxStreams
+          status1.getCode shouldBe Code.OK
+          status2.getCode shouldBe Code.OK
+          metrics.daml.lapi.streams.active.getCount shouldBe 0
+          status3.getCode shouldBe Code.ABORTED
+          status3.getDescription should include(metrics.daml.lapi.streams.activeName)
+          status4.getCode shouldBe Code.OK
+        }
+      }
+    }
+  }
+
+  it should "exclude non-stream traffic from stream counts" in {
+
+    val metrics = new Metrics(new MetricRegistry)
+
+    val limitStreamConfig = RateLimitingConfig.Default.copy(maxStreams = 2)
+
+    val waitService = new WaitService()
+    withChannel(metrics, waitService, limitStreamConfig).use { channel: Channel =>
+      {
+        for {
+
+          fStatus1 <- streamHello(channel)
+          fHelloStatus1 = singleHello(channel)
+          fStatus2 <- streamHello(channel)
+          fHelloStatus2 = singleHello(channel)
```

Comment:
I would add another `streamHello` right above to ascertain that a stream is rejected where a single is allowed